### PR TITLE
[Feat] - Updated Docusaurus to v2.4.1,  Added New Workflows and Migrated to TS

### LINKED ISSUE #90: [CI] Deploy CI should be triggered when a PR is raised
### Is your feature request related to a problem? Please describe.

It happened several times that I found the Deploy CI failed after the PR was merged. This requires the contributor to raising another PR to fix CI, which is not efficient.

### Describe the solution you'd like.

When a new PR is raised, the Deploy CI should be triggered. So the contributor can know whether their code works.

### Describe alternatives you've considered

_No response_

### Additional context.

_No response_

### LINKED ISSUE #92: Bug: Package file has duplicate dev dependencies
### Describe the bug.

The package.json file has duplicate devDependency of `"@docusaurus/module-type-aliases": "^2.4.0",`

### Describe the steps to reproduce the behavior.

- Go to https://github.com/WasmEdge/docs/blob/main/package.json
- Go to L43
- Go to L48
- L43 & L48 have the same dependencies which need to be removed

### Expected behavior.

Duplicate devDependencies should be removed

### Screenshots.

Screenshot 2023-06-05 at 10 15 29 PM

### Operating Environment

macOS

### Additional Information

Issue came from https://github.com/WasmEdge/docs/pull/71

### PR DESCRIPTION
**Pull request by adithyaakrishna**
**Description:**

- This PR updates docusaurus to v2.4.1
- Adds new workflows for lint, dependency review, and build check
- Updates old workflows
- Fixes #90 
- Fixes #92 
- PR migrates the repo from JS to TS

### CODE REVIEW

**Comment by adithyaakrishna** [comment]
@alabulei1 This PR adds a new workflow i.e. the Build Check to make sure the PR doesnt break the builds before they are merged

File: .github/workflows/build-test.yml [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+            - '.husky/**'
+            - '.env'
+            - '.gitignore'
+
+concurrency:
+    group: ${{ github.workflow }}-${{ github.head_ref || github.run_id }}
+    cancel-in-progress: true
+
+jobs:
+    Build:
```
Comment (adithyaakrishna) [inline .github/workflows/build-test.yml:30]: @alabulei1 This is the check for the build of the docs repo. It also fixes #90

**Comment by adithyaakrishna** [comment]
@alabulei1 Done, this PR is ready for review 🚀

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -8,35 +8,28 @@ If you are ready to contribute, please search for open issues labeled "help-want
...
-  - <a href="#commit-signoff">Developer Certificate of Origin</a>
-- WasmEdge Contribution Flow
-  - <a href="#contributing-docs">WasmEdge Docs</a> 
-  - <a href="https://wasmedge.org/book/en/contribute/contribute.html">WasmEdge Runtime</a>
-  - <a href="https://github.com/WasmEdge/WasmEdge/blob/master/SECURITY.md">Reporting a vulnerability</a>
+-   <a name="contributing-flow">Contribution Flow</a>
+    -   <a href="#commit-signoff">Developer Certificate of Origin</a>
+-   WasmEdge Contribution Flow
+    -   <a href="#contributing-docs">WasmEdge Docs</a>
+    -   <a href="https://wasmedge.org/book/en/contribute/contribute.html">WasmEdge Runtime</a>
```
Comment (alabulei1) [inline CONTRIBUTING.md:15]: This link should be the new docs, which you're working on.

File: docs/contribute/release.md [hunk 3]
```diff
@@ -6,62 +6,62 @@ sidebar_position: 9
...
-- [ ] Update `WASMEDGE_CAPI_VERSION` in `CMakeLists.txt`.
-- [ ] Update `wasmedge_version` in `docs/book/en/book.toml`.
-- [ ] Use git tag to create a new release tag ``major.minor.patch-rc.version``. And push this tag to GitHub.
-- [ ] Wait for the CI builds and pushes the release binaries and release notes to the GitHub release page.
-- [ ] Check the ``Pre-release`` checkbox and publish the pre-release.
-- [ ] This step will automatically close and announce the official release in about 3 days if there's no critical issues.
+-   [ ] In this step, the issue fixing is finished. The `RC` pre-releases are for the installation, bindings, and packages testing.
+-   [ ] Make sure that all the issues in the releasing process issue are completed.
+-   [ ] Update `WASMEDGE_CAPI_VERSION` in `CMakeLists.txt`.
+-   [ ] Update `wasmedge_version` in `docs/book/en/book.toml`.
```
Comment (alabulei1) [inline docs/contribute/release.md:43]: This should be the new docs

File: docs/develop/rust/dapr.md [hunk 4]
```diff
@@ -16,36 +16,36 @@ Before we started, make sure [you have Rust and WasmEdge installed](setup).
...
 
-The [classify](https://github.com/second-state/dapr-wasm/tree/main/image-api-classify) microservices takes an image from an HTTP POST, runs a Tensorflow model against it to classify the object on the image, and returns the result as a text label in the HTTP response. You can learn more about AI inference in Rust and WasmEdge [here](https://wasmedge.org/book/en/write_wasm/rust/wasinn.html). It uses its own Dapr sidecar the same way as the [grayscale](https://github.com/second-state/dapr-wasm/tree/main/image-api-grayscale) microservice. 
+The [classify](https://github.com/second-state/dapr-wasm/tree/main/image-api-classify) microservices takes an image from an HTTP POST, runs a Tensorflow model against it to classify the object on the image, and returns the result as a text label in the HTTP response. You can learn more about AI inference in Rust and WasmEdge [here](https://wasmedge.org/book/en/write_wasm/rust/wasinn.html). It uses its own Dapr sidecar the same way as the [grayscale](https://github.com/second-state/dapr-wasm/tree/main/image-api-grayscale) microservice.
```
Comment (alabulei1) [inline docs/develop/rust/dapr.md:35]: This also should be the new docs.

File: docs/contribute/installer.md [hunk 5]
```diff
@@ -49,115 +48,115 @@ The real installer handles all stuff. It supports python2.7 (not tested on earli
...
+-   Note - Currently `--plugins` is an experimental option.
 
-* Full Option: `--plugins wasi_crypto:0.11.0`
+-   Full Option: `--plugins wasi_crypto:0.11.0`
 
-* Note - The format for this argument is `<plugin_name>:<version_number>`. `<version_number>` is not compulsory. For example `--plugins wasi_crypto` is a valid option.
-* Note - `<plugin_name>` is cases sensitive. Allowed values are stated [here](https://wasmedge.org/book/en/plugin.html) in the `Rust Crate` column. The logic is that the release name should be the same.
-* Note - It's the same as the WasmEdge version if not specified.
+-   Note - The format for this argument is `<plugin_name>:<version_number>`. `<version_number>` is not compulsory. For example `--plugins wasi_crypto` is a valid option.
+-   Note - `<plugin_name>` is cases sensitive. Allowed values are stated [here](https://wasmedge.org/book/en/plugin.html) in the `Rust Crate` column. The logic is that the release name should be the same.
```
Comment (alabulei1) [inline docs/contribute/installer.md:129]: The link should be the new docs.

**Review by alabulei1 — APPROVED** [review]

**Comment by alabulei1** [comment]
Thanks @adithyaakrishna 

Could you tell me why `-` is better than `*`?

**Comment by adithyaakrishna** [comment]
> Thanks @adithyaakrishna
> 
> Could you tell me why `-` is better than `*`?

Not particulary, they both are used to denote lists itself. But GitHub Flavoured markdown uses `-`. 

For example, if you click the lists option here, it adds a `-`, I wanted to make it consistent across the repo, hence used `-`
Screenshot 2023-06-07 at 9 38 22 AM

File: docs/contribute/installer.md (hunk 5 above)
Comment (adithyaakrishna) [reply docs/contribute/installer.md:129]: @alabulei1 Could you please let me know if this is the correct link? https://wasmedge.org/docs/category/wasmedge-plugin-system

File: docs/contribute/release.md (hunk 3 above)
Comment (adithyaakrishna) [reply docs/contribute/release.md:43]: Done ✅

File: CONTRIBUTING.md (hunk 2 above)
Comment (adithyaakrishna) [reply CONTRIBUTING.md:15]: Done ✅

File: docs/develop/rust/dapr.md (hunk 4 above)
Comment (adithyaakrishna) [reply docs/develop/rust/dapr.md:35]: Done ✅

File: docs/contribute/installer.md (hunk 5 above)
Comment (alabulei1) [reply docs/contribute/installer.md]: Yes it is.

**Comment by alabulei1** [comment]
LGTM!

**Pull request merged**